packages/pwa-buildpack/docs/PWADevServer.md

```diff
@@ -77,19 +73,80 @@ configuration.
-- `serviceWorkerFileName: string`: **Required.** The name of the ServiceWorker
-   file this project generates, e.g. `'sw.js'`.
-- `changeOrigin: boolean`: ⚠️ **(experimental)** Try to parse any HTML responses
-   from the proxied Magento backend, and replace its domain name with the
-   dev server domain name. Default `false`.
+   backend server, e.g. `'/'`.
+- `provideSecureHost: boolean | SecureHostOptions`: Use a [secure and unique hostname for the dev server.](#securehostoptions)
+- `graphqlPlayground: GraphQLPlaygroundOptions`: Add a [pre-populated GraphQL Playground to the dev server.](#graphqlplayground)
+- `id: string`: :no_entry_sign: **_Deprecated._** A unique ID for this project. This id will be used to create your dev domain name. Setting `id: 'foo'` is equivalent to:
+
+  ```js
+  provideSecureHost: { subdomain: 'foo', addUniqueHash: false }
+  ```
+
+#### `SecureHostOptions`
+
+Most local development servers run on `http://localhost`, with a constant port
+number like `3000` or `8080`. This common setup is problematic for PWA work:
+
+- Browsers only enable PWA features like ServiceWorkers on secure HTTPS.
+- Browsers cache PWAs by "scope", which is an origin plus a path (usually `/`).
+  Running all local development projects at a common domain, such as
+  `https://localhost:8000`, will cause conflicts between ServiceWorkers.
+  
+To solve these problems, enable autogeneration of a unique secure host using the
+`provideSecureHost` configuration option. PWADevServer generates the unique
+hostname based on the project name in `package.json` if available, or a
+custom name if provided. It also uses a hash of the filesystem path of the
+project to choose a persistent unique port and append a short random sequence
+to the domain name. Since this port and string are derived from the
+filesystem path, they will stay the same unless the project is moved.
+
+:information_source: If PWADevServer detects that the project port is in use,
+it will print a warning and use a different port temporarily.
+
+Configuration options for this feature are:
+
+- `subdomain: string`: A custom subdomain string to use. If provided, this
+  supersedes the name in `package.json`.
+- `exactDomain: string` A fully qualified domain to use. By default,
+  PWADevServer generates all unique hosts as subdomains of `local.pwadev`. Use
+  this option to override this behavior and provide the exact domain to use.
+  The custom host will not have a unique hash, but it will have a unique port.
+- `addUniqueHash: boolean`: Use the filesystem path hash to create a short
+  string of URL-safe characters to append to the subdomain. Ensures total
+  uniqueness of domain. Default `true`. If `exactDomain` is specified, this
+  option has no effect.
+
+:information_source: Default behavior is to use the `package.json` plus hash
+as a subdomain of `local.pwadev`. To use this default behavior, simply set
+`provideUniqueHost: true`.
```

Comment:
Is this supposed to be `provideSecureHost` instead of `provideUniqueHost`?

Reply:
Yes, whoops!

Reply:
```suggestion
`provideSecureHost: true`.
```

Trying the new suggestions feature...